lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -72,10 +72,19 @@ public void close() {
 	public class Channels extends HashMap<String, Channel> {
 		public Channels() {
 			/* remove all channels when the connection is closed, to avoid stalled state */
-			connection.on(ConnectionState.closed, new ConnectionStateListener() {
+			connection.on(new ConnectionStateListener() {
```

Comment:
Thank for your comment, but I'm not sure that I'm well understand, that the listener will be removed, do you mean I should remove all listeners?